internal/internal_poller_autoscaler.go

```diff
@@ -139,9 +150,12 @@ func (p *pollerAutoScaler) Start() autoscaler.DoneFunc {
 			}
 		}
 	}()
-	return func() {
-		p.cancel()
-	}
+	return
+}
+
+// Stop stops the poller autoscaler
+func (p *pollerAutoScaler) Stop() {
+	p.cancel()
```

Comment:
not relevant to this PR, but in general: we probably want a (time-bounded?) blocking "... and wait for shutdown to complete" behavior here.
otherwise it's impossible to rule out dangling goroutines (e.g. goleak tests) or race conditions (background loop triggering at the same time as cancelation).